modules/axtask/src/run_queue.rs

```diff
@@ -282,9 +299,7 @@ impl<'a, G: BaseGuard> CurrentRunQueueRef<'a, G> {
         trace!("task migrate: {}", curr.id_name());
         assert!(curr.is_running());
 
-        select_run_queue::<NoOp>(curr.as_task_ref())
-            .inner
-            .put_task_with_state(curr.clone(), TaskState::Running, false);
+        curr.set_state(TaskState::Migrating);
 
         self.inner.resched();
```

Comment:
Maybe we can introduce a new function `migrate_resched` to reduce the interference of the regular rescheduling.

Reply:
We can just select run queue and put the previous task after `resched`.

Reply:
you mean, let "next task" to put previous task into run queue?

Reply:
Yes, move `finish_prev_task_switch ` after `self.inner.resched()`, and leave `clear_prev_task_on_cpu` in `switch_to`.

Reply:
> Yes, move `finish_prev_task_switch ` after `self.inner.resched()`, and leave `clear_prev_task_on_cpu` in `switch_to`.

`finish_prev_task_switch` can not be placed after `self.inner.resched()`, because it should be the next task that is responsible for finishing the scheduling process of previous task.

I still combined `clear_prev_task_on_cpu` into `finish_prev_task_switch`, and put it in `switch_to`